lib/ansible/modules/system/aix_devices.py

```diff
@@ -193,16 +193,17 @@ def discover_device(module, device):
     """ Discover AIX devices."""
     cfgmgr_cmd = module.get_bin_path('cfgmgr', True)
 
-    if device is not None:
-        device = "-l %s" % device
-
+    # device is mandatory according to doc spec,
+    #  if  device == 'all', run bare cfgmgr.  else, scan specific device
+    if device.lower() == 'all':
```

Comment:
`device` is defined as Required in the parameters [doc spec](https://docs.ansible.com/ansible/latest/modules/aix_devices_module.html#parameters), so it should never be None.  

Having said that, there's no specification that devices must be lower-case, and AIX itself uses case sensitive device names.  We could remove it entirely just in case someone renames a device `ALL` and we force them to run `cfgmgr` when they try to modify it.